src/GrpAb/GrpAbFinGen.jl

```diff
@@ -77,38 +78,21 @@ function abelian_group(M::fmpz_mat; name::String = "")
   else
     G = GrpAbFinGen(M)
   end
-  if name != ""
-    set_name!(G, name)
-  end
+  name == "" || set_name!(G, name)
   return G
 end
 
 @doc Markdown.doc"""
-    abelian_group(M::Matrix{fmpz}) -> GrpAbFinGen
+    abelian_group(M::Matrix{T}) where T <: Union{Integer, fmpz} -> GrpAbFinGen
```

Comment:
OK, I just realized I got this wrong, it should be this (of course also elsewhere):
```suggestion
    abelian_group(M::Matrix{<:T}) where T <: Union{Integer, fmpz} -> GrpAbFinGen
```
or even use `<:AbstractMatrix{<:T}` 

Also, perhaps writing it like this would be more userfriendly?
```suggestion
    abelian_group(M::Matrix{fmpz}) -> GrpAbFinGen
    abelian_group(M::Matrix{<:Integer}) -> GrpAbFinGen
```
But then again, if we introduce `Nemo.IntegerUnion`, ew could just use that and write
```suggestion
    abelian_group(M::<AbstractMatrix{<:IntegerUnion}) -> GrpAbFinGen
```

Reply:
I believe the docstrings are allowed to be sloppy so I would also allow
`Matrix{<:Union{Integer, Union}}` or `AbstractMatrix{<:Union{Integer, Union}}`. Not sure if this shared by everyone.

Reply:
Err, yeah, I messed up some of these, but I hope the gist is clear ;-)

Reply:
I've updated it

Reply:
The first is also not wrong, since `T <: Union{S, ...}` also allows all `T` with `T <: S`.